src/Tools/babylon.tools.ts

```diff
@@ -3,6 +3,16 @@
         animations: Array<Animation>;
     }
 
+    // See https://stackoverflow.com/questions/12915412/how-do-i-extend-a-host-object-e-g-error-in-typescript

```

Comment:
I would prefer not inheriting from error and wrapping the original error for callstack. 

Also do not rely on __proto__ on your on you can use setPrototypeOf in Typescript

Reply:
> I would prefer not inheriting from error and wrapping the original error for callstack.

I tried that, but it has issues. Exceptions that are throw will no longer look right in the browser. It will just say the name of the class.

> Also do not rely on proto on your on you can use setPrototypeOf in Typescript

Object.setPrototypeOf is not available in the lib.es5.d.ts. Supposedly, it also does not work in IE11. Perhaps I can try to call it if it exists?

Reply:
Yup, so should we remove the inheritance ? As setting proto might be pretty slow and lead to deoptim.

Reply:
Just tested it and Object.setPrototypeOf does work in IE11.

> should we remove the inheritance?

My preference is to use it since it works much better with the debugger. Someone might be checking to see if the exception is an instance of Error, in which case not inheriting from Error would be broken.

> As setting proto might be pretty slow and lead to deoptim.

I'm not sure this is a big issue. This type of exceptions should be rare and far in between.

Reply:
I've updated the code to this locally:

```javascript
            if ((Object as any).setPrototypeOf) {
                (Object as any).setPrototypeOf(this, LoadFileError.prototype);
            }
            else {
                (this as any).__proto__ = LoadFileError.prototype;
            }
```